cmd/snap/error.go

```diff
@@ -190,3 +220,136 @@ If you understand and want to proceed repeat the command including --classic.
+		}
+	}
+
+	if kind == client.ErrorKindRevisionNotAvailableForArchitecture {
+		// TODO: add "Get more information..." hints once snap info
+		// support showing multiple/all archs
+
+		if hits := matches["track:risk"]; len(hits) != 0 {
+			archs := strings.Join(archsForChannels(hits), ", ")
+			msg := fmt.Sprintf(i18n.G("snap %q is not available on %v for this architecture (%s) but exists on other architectures (%s)."), snapName, req, arch, archs)
+			return msg
+		}
+
+		archs := strings.Join(archsForChannels(avail), ", ")
+		msg := fmt.Sprintf(i18n.G("snap %q is not available on this architecture (%s) but exists on other architectures (%s)."), snapName, arch, archs)
+		return msg
+	}
+
+	if len(matches["architecture:track:risk"]) != 0 && req.Branch != "" {
+		trackRisk := snap.Channel{Track: req.Track, Risk: req.Risk}
+		trackRisk = trackRisk.Clean()
+		msg := fmt.Sprintf(i18n.G("requested an apparently non-existing branch on %s for snap %q: %s"), trackRisk.Full(), snapName, req.Branch)
+		return msg
+	}
+
+	moreInfoHint := fmt.Sprintf(i18n.G("Get more information with 'snap info %s'."), snapName)
+	preRelWarn := i18n.G("Please be mindful pre-release channels may include features not completely tested or implemented.")
+	trackWarn := i18n.G("Please be mindful that different tracks may include different features.")
+
+	if hits := matches["architecture:track"]; len(hits) != 0 {
+		msg := fmt.Sprintf(i18n.G("snap %q is not available on %v but is available to install on the following channels:\n"), snapName, req)
+		msg += installTable(snapName, action, hits, false)
+		msg += "\n"
+		if req.Risk == "stable" {
+			msg += "\n" + preRelWarn
+		}
+		msg += "\n" + moreInfoHint
+		return msg
+	}
+	if hits := matches["architecture:risk"]; len(hits) != 0 {
+		msg := fmt.Sprintf(i18n.G("snap %q is not available on %s but is available to install on the following tracks:\n"), snapName, req.Full())
+		msg += installTable(snapName, action, hits, true)
+		msg += "\n\n" + trackWarn
+		msg += "\n" + moreInfoHint
+		return msg
+	}
+
+	msg := fmt.Sprintf(i18n.G("snap %q is not available on %s but other tracks exist.\n"), snapName, req.Full())
+	msg += "\n\n" + trackWarn
+	msg += "\n" + moreInfoHint
+	return msg
+}
+
+func installTable(snapName, action string, avail []*snap.Channel, full bool) string {
+	b := &bytes.Buffer{}
+	w := tabwriter.NewWriter(b, 10, 3, 2, ' ', 0)
+	first := true
+	for _, a := range avail {
+		if first {
+			first = false
+		} else {
+			fmt.Fprint(w, "\n")
+		}
+		var ch string
+		if full {
+			ch = a.Full()
+		} else {
+			ch = a.String()
+		}
+		chOption := channelOption(a)
+		fmt.Fprintf(w, "%s\tsnap %s %s %s", ch, action, chOption, snapName)
+	}
+	w.Flush()
+	tbl := b.String()
+	// indent to drive fill/ToText to keep the tabulations intact
+	lines := strings.SplitAfter(tbl, "\n")
+	for i := range lines {
+		lines[i] = "  " + lines[i]
+	}
+	return strings.Join(lines, "")
+}
+
+func channelOption(c *snap.Channel) string {
```

Comment:
wouldn't this be just String of the channel, provided it's normalised?

Reply:
how can it be given that is really an option with -- and channel= etc?  also normalisation follows the store rules

so "latest/stable" is "stable"  but  "1.0/stable" is "1.0/stable" not 1.0 which we can use here

Reply:
yep, you're right.  Sorry.